### PR TITLE
SWAP-1201 CI/CD Improvements

### PR DESCRIPTION
**Pull request by peter-asztalos**
## Description

This PR updates the Github action and Docker configurations to improve the overall speed, size of the images and makes action files consistent.

## Motivation and Context

The previous solution didn't use any caching and the generated image was large.

## Fixes

https://jira.esss.lu.se/browse/SWAP-1201

## Changes

- save intermediate layers after each build and use them if there were no changes
- docker ignore updated to provide a better caching experience 
- use multi-stage builds and alpine to reduce image size (~1gb->~480mb)
- package.json updated to use only proper dependencies in each environment
- e2e test related configurations

## Depends on

https://github.com/UserOfficeProject/user-office-factory/pull/6

## Tests included/Docs Updated?

- [ ] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

**Comment by peter-asztalos** [comment]
@UserOfficeProject/stfc-business-apps

File: src/__tests__/sample-pdf.spec.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import request from 'supertest';
+
+import extractPDFText from '../../scripts/text-extraction';
+import app from '../app';
+import { getTotalPages } from '../pdf';
+import { generateTmpPath } from '../util/fileSystem';
+import testPayloads from './payloads.json';
+
+beforeAll(done => {
+  setTimeout(done, 5000);
```
Comment (jekabs-karklins) [inline src/__tests__/sample-pdf.spec.ts:12]: Why is the timeout needed?

**Review by jekabs-karklins — COMMENTED** [review]
I see you have added code changes for PDF generation in SWAP-1204, is that intended?

**Comment by peter-asztalos** [comment]
> I see you have added code changes for PDF generation in SWAP-1204, is that intended?

SWAP-1204 is about Helm charts, not sure what you mean.

File: src/__tests__/sample-pdf.spec.ts (hunk 1 above)
Comment (peter-asztalos) [reply src/__tests__/sample-pdf.spec.ts:12]: It's for warmup, like letting puppeteer launch and other startup scripts to run.

**Comment by jekabs-karklins** [comment]
> > I see you have added code changes for PDF generation in SWAP-1204, is that intended?
> 
> SWAP-1204 is about Helm charts, not sure what you mean.

I mean, that there seems to be something wrong with the commits from this pull request as they also include not only SWAP-1204 related commits about updating  helm charts but also about PDF generation.
See here https://github.com/UserOfficeProject/user-office-factory/pull/7/files

**Comment by peter-asztalos** [comment]
Oh, I see, it's because it depends on https://github.com/UserOfficeProject/user-office-factory/pull/6 which wasn't merged yet.

**Review by vovsike — APPROVED** [review]

**Pull request merged**